### ISSUE TITLE
Let users know that the input length too short/long without waiting in the queue in `gr.Audio` and `gr.Video`

### ISSUE THREAD

**Issue by hysts** [gradio-app/gradio#6775]
- [x] I have searched to see if a similar issue already exists.

I'm wondering if it's possible to let users know that the input audio/video length is too short/long without waiting in the queue. I mean, my understanding is that users get the error that says it's too short/long after waiting in the queue with the current implementation, but I don't think it's a good UX.

[internal discussion](https://huggingface.slack.com/archives/C02QZLG8GMN/p1702481510558099)

https://github.com/gradio-app/gradio/blob/1459119c6eb43cf9780ffb2b97db845bd917fe3e/gradio/components/audio.py#L196-L203
https://github.com/gradio-app/gradio/blob/1459119c6eb43cf9780ffb2b97db845bd917fe3e/gradio/components/video.py#L173-L180

**Comment by pngwn** [comment]
We should emit and error or warning in the UI when the event is dispatched. The challenge here is that only the component knows if a value is valid but it doesn't necessarily know when an even it being dispatched. 

I think it would be best if validators were run by the core architecture but provided by the component. This would be a generic interface which would avoid custom validation implementations for every component. 

I propose that each component can optionally export a `validator` function which the data will be passed through before dispatching an event. The validator function should accept the data as an argument, it already has access to all props if needed, it should return a Boolean: true if the check is passed, false if not. We can check to see if this function is exported , if it is then we use it, if not then we just proceed as normal.

**Comment by dawoodkhan82** [comment]
@pngwn makes sense! I can tackle this. Should we show a `gr.Warning` for invalid inputs? Also what about for our form elements, the simple css built-in `out-of-range` selector should be sufficient.

**Comment by pngwn** [comment]
The event won't submit at all so maybe it should be an error? I don't think it matters too much though, as long as they are notified about it immediately. 

If we already have built in validation then I think it is fine to use that, as long as it prevents the event from being triggered and is obvious to the user.

**Comment by pngwn** [comment]
We have some similar logic to this for `get_value` so we can probably do something similar.

**Comment by freddyaboulton** [comment]
I think it would be useful as well if we added a python api for custom validation (#2718) that does not have to wait in the queue.

**Comment by pngwn** [comment]
I feel like we could take a similar approach. Eg, an optional validator method in the python class or something.

**Comment by freddyaboulton** [comment]
Closing in favor of #11875 thanks @hysts !
